experimental/dds/tree2/src/core/tree/cursor.ts

```diff
@@ -363,6 +363,36 @@ export function forEachNode<TCursor extends ITreeCursor = ITreeCursor>(
 	}
 }
 
+/**
+ * Moves cursor to each child of the current field.  The optional arguments `start` and `end`
+ * can be used to restrict the range of children visited.
+ *
+ * `callbackFn` is invoked after moving the cursor to each child.  The callback must return
+ * `true` to continue the enumeration, or `false` to early exit.
+ *
+ * The `callbackFn` may move move the provided cursor, but must restore the cursor to the current
+ * child node in the enumeration before returning.
+ */
+export function forEachNodeIn<TCursor extends ITreeCursor = ITreeCursor>(
+	cursor: TCursor,
+	callbackFn: (cursor: TCursor) => boolean,
+	start = 0,
+	end = cursor.getFieldLength(),
+): void {
+	assert(cursor.mode === CursorLocationType.Fields, 0x3bd /* should be in fields */);
```

Comment:
this will fail the assert tagging policy check. DO not copy paste assert tags.
```suggestion
	assert(cursor.mode === CursorLocationType.Fields, "should be in fields");
```

Reply:
Thanks.  Glad that policy-check would have caught it.